### PR TITLE
New adding field type UI

### PR DESCRIPTION
**Pull request by guyyoo**

### CODE REVIEW

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 1]
```diff
@@ -72,108 +81,156 @@ export default function (AB) {
...
-               cols: [
+               paddingX: 2,
+               elements: [
                   {
                      view: "label",
-                     label: L("Add new field"),
-                     css: "modal_title",
+                     id: ids.title,
                      align: "center",
+                     label: L("<span>Choose Field-Type</span>"),
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:96]: Do we need the <span> tag or can it be removed?

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 2]
```diff
@@ -72,108 +81,156 @@ export default function (AB) {
...
-                        onAfterRender() {
-                           UIClass.CYPRESS_REF(this);
-                        },
-                     },
-                  },
+                     cols: [
+                       {
+                           view: "button",
+                           id: ids.buttonMaximize,
+                           label: L("<span class=\"webix_icon\"><i class=\"nomargin fa fa-expand\"></i></span>"),
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:104]: I think if we are not going to translate this to another language you do not have to wrap in the L() function.

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 3]
```diff
@@ -72,108 +81,156 @@ export default function (AB) {
...
+                           css: "webix_transparent",
+                           width: 40,
+                           click: () => {
+                              this.buttonMaximize();
+                           }
+                       },
+                       {
+                           view: "button",
+                           id: ids.buttonMinimize,
+                           label: L("<span class=\"webix_icon\"><i class=\"nomargin fa fa-compress\"></i></span>"),
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:114]: I think if we are not going to translate this to another language you do not have to wrap in the L() function.

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 4]
```diff
@@ -72,108 +81,156 @@ export default function (AB) {
...
+                           hidden: true,
+                           css: "webix_transparent",
+                           width: 40,
+                           click: () => {
+                              this.buttonMinimize();
+                           }
+                       },
+                       {
+                           view: "button",
+                           label: L("<span class=\"webix_icon\"><i class=\"nomargin fa fa-times\"></i></span>"),
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:124]: I think if we are not going to translate this to another language you do not have to wrap in the L() function.

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 5]
```diff
@@ -72,108 +81,156 @@ export default function (AB) {
...
-                              this.onChange(id);
-                           },
+               rows: [
+                  {
+                     id: ids.chooseFieldType,
+                     rows: [
+                        {
+                           view: "search",
+                           id: ids.searchBar,
+                           placeholder: "Search by title...",
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:152]: Wrap this in the L() function so we can translate it later... 

`placeholder: L("Search by title..."),`

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 6]
```diff
@@ -213,8 +270,11 @@ export default function (AB) {
             var menuName = F.defaults().menuName;
             var key = F.defaults().key;
 
+            const icon = F.defaults().icon;
+
             // add a submenu for the fields multilingual key
-            this.submenus.push({ id: menuName, value: L(menuName) });
+            // this.submenus.push({ id: menuName, value: L(menuName) });
+            this.submenus.push({ id: menuName, icon: `nomargin fa fa-${icon}`,  labelNickName: L(menuName), label: L(menuName)});
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:277]: labelNickName and label seem to be the same, do we use them in different ways?

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 7]
```diff
@@ -632,14 +719,22 @@ export default function (AB) {
...
        */
-      onChange(name) {
+
+      onClick(name) {
+         // show Field Type popup
+         $$(this.ids.chooseFieldType).hide();
+         $$(this.ids.fieldSetting).show();
+
+         // set title name by each field type
+         $$(this.ids.title).setValue(L(`<span>Create Field: ${name}</span>`));
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:729]: We can translate the first part but the "name" will have its translation stored elsewhere so try this...

`$$(this.ids.title).setValue(L("Create Field:") + " " + name");`

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 8]
```diff
@@ -675,7 +770,47 @@ export default function (AB) {
...
+         this.buttonMinimize();
+
+          // show the ability to switch data types
+         $$(this.ids.chooseFieldType).show();
+
+         // show button "Back"
+         $$(this.ids.buttonBack).show();
+
+         // set title name to "Choose Field-Type"
+         $$(this.ids.title).setValue(L(`<span>Choose Field-Type</span>`));
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:786]: Remove span tag.

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 9]
```diff
@@ -675,7 +770,47 @@ export default function (AB) {
...
+
+         // hide form editor
+         $$(this.ids.fieldSetting).hide();
+
+         // change button text to 'Create'
+         $$(this.ids.buttonSave).define("label", L("<span class=\"text\">Create</span>"));
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:792]: Remove span tag.

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 10]
```diff
@@ -675,7 +770,47 @@ export default function (AB) {
...
+         this.buttonMinimize();
+
+         // hide the ability to switch data types
+         $$(this.ids.chooseFieldType).hide();
+         
+         // hide button "Back"
+         $$(this.ids.buttonBack).hide();
+
+         // set title name by each field type
+         $$(this.ids.title).setValue(L(`<span>Create Field: ${fieldTypeName}</span>`));
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:806]: Remove span tag.

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js [hunk 11]
```diff
@@ -675,7 +770,47 @@ export default function (AB) {
...
+
+         // show form editor
+         $$(this.ids.fieldSetting).show();
+
+         // change button text to 'save'
+         $$(this.ids.buttonSave).define("label", L("<span class=\"text\">Save</span>"));
```
Comment (jamesduncan) [inline src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:812]: Remove span tag.

File: src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js (hunk 1 above)
Comment (jamesduncan) [reply src/rootPages/Designer/ui_work_object_workspace_popupNewDataField.js:96]: label: L("Choose Field-Type"),
label: "<span>" + L("Chooose Field-Type") + "</span>",

**Review by jamesduncan — CHANGES_REQUESTED** [review]

**Review by jamesduncan — APPROVED** [review]

**Review by nh758 — APPROVED** [review]

**Pull request merged**